workflow-templates/static-analysis.yml

```diff
@@ -60,3 +60,5 @@ jobs:
 
       - name: Psalm
         uses: docker://vimeo/psalm-github-actions
+        with:
+          composer_require_dev: true
```

Comment:
Is this something we are going to need more often than not?

Reply:
mm I don't know for sure, so far it has been always added (just 2 😬 ): https://github.com/search?q=org%3Adoctrine+docker%3A%2F%2Fvimeo%2Fpsalm-github-actions&type=code, shall I just comment it?

Reply:
Yeah commenting out might be better, as well as adding a comment showing when this is likely to be needed (why was it needed so far?)

Reply:
By "why was it needed so far" I mean: to understand what? The PHPUnit API? If that's the case then we are likely to always need this, are we not?

Reply:
Sorry I didn't understand the comment 😬 In the case of `doctrine/persistence` (the one I added) it was needed because [`RuntimePublicReflectionProperty.php`](https://github.com/doctrine/persistence/blob/6236bc22a42481e190f2df3e7a8fb233247acba6/lib/Doctrine/Persistence/Reflection/RuntimePublicReflectionProperty.php#L5) and [its test](https://github.com/doctrine/persistence/blob/6236bc22a42481e190f2df3e7a8fb233247acba6/tests/Doctrine/Tests/Persistence/RuntimePublicReflectionPropertyTest.php#L6) are  using `Doctrine\Common\Proxy\Proxy` (`doctrine/common`): https://github.com/doctrine/persistence/actions/runs/369287285.

Related: https://github.com/doctrine/persistence/pull/122

Reply:
Oh ok so it's really needed when we want to test with optional dependencies, things like that. Makes sense!

Reply:
Can someone please explain to me why it's a comment now? I don't understand why it isn't kept in the config.

Reply:
I think (but I'm not sure), that it's not required in most situations (because Psalm comes with stubs?) The situations where you will need this is when you are testing a Doctrine project with another that is in require-dev (I have seen `doctrine/orm` in `require-dev` of some packages, for instance).